src/Hosting.Services.Remoting/RemotingListenerBuilder.cs

```diff
@@ -20,23 +21,21 @@ public abstract class RemotingListenerBuilder<TService> : IListenerBuilder<TServ
 		public string Name { get; }
 
 		// done internal for unit tests
-		internal FabricTransportRemotingListenerSettings? Settings { get; }
+		internal FabricTransportRemotingListenerSettings Settings { get; }
 
 		/// <summary>
 		/// Constructor
 		/// </summary>
+		/// <remarks>Transport security settings should be defined in TransportSettings of a service's settings.xml before creating a listener</remarks>
 		protected RemotingListenerBuilder(
 			string name,
 			FabricTransportRemotingListenerSettings? settings = null)
 		{
-			if (settings == null)
+			if (settings == null && !FabricTransportRemotingListenerSettings.TryLoadFrom("TransportSettings", out settings))
 			{
-				if (!FabricTransportRemotingListenerSettings.TryLoadFrom("TransportSettings", out settings))
-				{
-					settings = new FabricTransportRemotingListenerSettings();
-				}
+				throw new InvalidOperationException("Transport security is required for Service Fabric Remoting connections. TransportSettings must be defined in settings.xml.");
```

Comment:
Probably not worth it, but given that this message will be used twice in the codebase, would it be better to define a constant?

Reply:
I debated creating a custom exception type to unify it but I don't think it matters that much.
Given that you've commented on it too, maybe I should add it.

Reply:
Added a custom exception type.